bach/bach/series/series_json.py

```diff
@@ -465,6 +474,37 @@ def _get_slice_partial_expr(self, value: Optional[int], is_start: bool) -> Expre
             array_len = self.get_array_length()
             return Expression.construct(f'({{}} {value})', array_len)
 
+    def _find_in_json_list(self, filtering_slice: Dict[str, str], is_start: bool):
+        """
+        Return expression for the position of the element that contains the filtering values
+
+        Assumes that self._series_object is an array and each element is a json!
+
+        :param filtering_slice: dictionary containing the keys and values to use on filter.
+        :param is_start: whether value is the slice.start (True) or slice.stop (False) value. If true,
+            the first element containing the values is returned, otherwise the last.
+        :return: Expression that gets the position of the element based on the slicing filters.
+        """
+        if not isinstance(filtering_slice, dict):
+            raise TypeError(f'key should be a dict, actual type: {type(filtering_slice)}')
+
+        # this way we can reuse code from the accesor without duplicating.
+        # 'element' will filled by iterating over the json array in the query below.
+        element_json = self._series_object.copy_override(
+            expression=Expression.construct('element'),
+        )
+
+        all_filters = [
+            element_json.json.get_value(filter_key, as_str=True) == str(filter_value)
+            for filter_key, filter_value in filtering_slice.items()
+        ]
+        slicing_mask = reduce(operator.and_, all_filters)
+        fmt = (
+            f"(select {'min' if is_start else 'max'}(case when {{}} then pos end) "
```

Comment:
```suggestion
            f"(select {'min' if is_start else '1 + max'}(case when {{}} then pos end) "
```
not fully tested, but something like this gives the expected `.loc`-like behaviour